zenodb.go

```diff
@@ -153,6 +153,7 @@ type DB struct {
 	requestedIterations  chan *iteration
 	coalescedIterations  chan []*iteration
 	closed               bool
+	followMx             sync.Mutex
```

Comment:
This protects the newStreamSubscriber map, right?  It's not obvious from the name.  It's easy to figure out if you see this mutex and wonder what it's for.  However, I could also see someone down the road modifying the newStreamSubscriber map without realizing there's an associated mutex they should be using.